### PR TITLE
Re-throw runtime errors when failing to parse 3D Tiles content.

### PR DESCRIPTION
**Pull request by bagnell**
Re-throws a `RuntimeError` in `Cesium3DTileset` when parsing content fails. Errors like 404s are still ignored.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Signed CLA is on file.

@bagnell, thanks for the pull request! Maintainers, we have a signed CLA from @bagnell, so you can review this at any time.

:warning: I noticed that [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CHANGES.md) has not been updated. If this change updates the public API in any way, fixes a bug, or makes any non-trivial update, please add a bullet point to `CHANGES.md` and comment on this pull request so we know it was updated. For more info, see the [Pull Request Guidelines]( https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines).

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

File: Source/Scene/Cesium3DTileset.js [hunk 1]
```diff
@@ -1261,7 +1264,13 @@ define([
         tile.contentReadyPromise.then(function() {
             removeFunction();
             tileset.tileLoad.raiseEvent(tile);
-        }).otherwise(removeFunction);
+        }).otherwise(function(error) {
+            removeFunction();
+            if (error instanceof RuntimeError) {
```
Comment (lilleyse) [inline Source/Scene/Cesium3DTileset.js]: Also look for `DeveloperError`.

**Comment by lilleyse** [comment]
Fixes https://github.com/AnalyticalGraphicsInc/cesium/issues/5158. If it's not too much trouble could you also implement a `tileFailed` event like discussed there?

**Comment by bagnell** [comment]
@lilleyse Updated and added a test.

Instead of throwing the error, an event is raised with the url and message. If there are no listeners, it logs the error to the console.

File: CHANGES.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ Change Log
 * Improved CZML Custom Properties sandcastle example [#6086](https://github.com/AnalyticalGraphicsInc/cesium/pull/6086)
 * Added `Plane.projectPointOntoPlane` for projecting a `Cartesian3` position onto a `Plane` [#6092](https://github.com/AnalyticalGraphicsInc/cesium/pull/6092) 
 * Added `Cartesian3.vectorProjection` for projecting one vector to another [#6093](https://github.com/AnalyticalGraphicsInc/cesium/pull/6093) 
+* Added `Cesium3DTileset.tileFailed` event that will be raised when a tile fails to load. The object passed to the event listener will have a url and message property. [#6088](https://github.com/AnalyticalGraphicsInc/cesium/pull/6088)
```
Comment (lilleyse) [inline CHANGES.md]: Mention that if there are no listeners failed tile error messages will be logged to the console.

**Comment by lilleyse** [comment]
Besides that `CHANGES.md` comment this looks good.

**Comment by bagnell** [comment]
@lilleyse Updated. Also updated the event doc.

**Pull request merged**